test/unit/gateways/trexle_test.rb

```diff
@@ -0,0 +1,528 @@
+require 'test_helper'
+
+class TrexleTest < Test::Unit::TestCase
+  def setup
+    @gateway = TrexleGateway.new(:api_key => 'THIS_IS_NOT_A_REAL_API_KEY')
+
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      :email => 'john@trexle.com',
+      :billing_address => address,
+      :description => 'Store Purchase',
+      :ip => '127.0.0.1'
+    }
+  end
+
+  def test_required_api_key_on_initialization
+    assert_raises ArgumentError do
+      TrexleGateway.new
+    end
+  end
+
+  def test_default_currency
+    assert_equal 'USD', TrexleGateway.default_currency
+  end
+
+  def test_money_format
+    assert_equal :cents, TrexleGateway.money_format
+  end
+
+  def test_url
+    assert_equal 'https://core.trexle.com/api/v1', TrexleGateway.test_url
+  end
+
+  def test_live_url
+    assert_equal 'https://core.trexle.com/api/v1', TrexleGateway.live_url
+  end
+
+  def test_supported_countries
+    assert_equal ["AD",
+ "AE",
```

Comment:
You might as well use the same construct that reads beautifully in the implementation! :)
```ruby
expected_supported_countries = %w(AD AE AT AU BD BE BG BN CA CH CY CZ DE DK EE EG ES FI FR GB
                                  GI GR HK HU ID IE IL IM IN IS IT JO KW LB LI LK LT LU LV MC
                                  MT MU MV MX MY NL NO NZ OM PH PL PT QA RO SA SE SG SI SK SM
                                  TR TT UM US VA VN ZA)
assert_equal expected_supported_countries, TrexleGateway.supported_countries